src/storage/src/hummock/store/version.rs

```diff
@@ -992,11 +992,15 @@ impl HummockVersionReader {
         key_range: TableKeyRange,
         options: ReadLogOptions,
     ) -> HummockResult<ChangeLogIterator> {
-        let change_log = if let Some(change_log) = version.table_change_log.get(&options.table_id) {
-            change_log.filter_epoch(epoch_range).collect_vec()
-        } else {
-            Vec::new()
+        let change_log = {
+            let table_change_logs = version.table_change_log().read();
+            if let Some(change_log) = table_change_logs.get(&options.table_id) {
+                change_log.filter_epoch(epoch_range).cloned().collect_vec()
```

Comment:
This cloned() is an additional cost introduced in this PR.

If multiple iter_log processes are running simultaneously, will the memory usage be substantial?

Reply:
cc @wenym1 , suggests that `iter_log` is executed less frequently and that this clone is acceptable.